pkg/spec/postgresql.go

```diff
@@ -225,10 +234,29 @@ func extractClusterName(clusterName string, teamName string) (string, error) {
 	if strings.ToLower(clusterName[:teamNameLen+1]) != strings.ToLower(teamName)+"-" {
 		return "", fmt.Errorf("name must match {TEAM}-{NAME} format")
 	}
+	if len(clusterName) > clusterNameMaxLength {
+		return "", fmt.Errorf("name cannot be longer than %d characters", clusterNameMaxLength)
+	}
+	if !serviceNameRegex.MatchString(clusterName) {
+		return "", fmt.Errorf("name must confirm to a valid service name (DNS-1035)")
```

Comment:
RFC 1035 is a  50+ pages document. Can you please specify the section that provides overview of the naming  conventions and thus the regexp ?